src/app/core/store/customer/basket/basket.effects.ts

```diff
@@ -197,6 +203,30 @@ export class BasketEffects {
     )
   );
 
+  /**
+   * Sets a desired delivery date at the current basket and each line item.
+   */
+  setBasketDesiredDeliveryDate$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(setBasketDesiredDeliveryDate),
+      mapToPayloadProperty('desiredDeliveryDate'),
+      concatMap(date =>
```

Comment:
Does the rxjs operator concatMap make sense for setting a basket delivery date? The stream would wait for setting a new desiredDeliveryDate property until the following basket service observable completed.